lib/content-services/src/lib/infinite-scroll-datasource/infinite-scroll-datasource.spec.ts

```diff
@@ -0,0 +1,161 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ContentPagingQuery } from '@alfresco/js-api';
+import { ScrollingModule } from '@angular/cdk/scrolling';
+import { Component, OnInit } from '@angular/core';
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { TranslateModule } from '@ngx-translate/core';
+import { from, Observable } from 'rxjs';
+import { ContentTestingModule } from '../testing/content.testing.module';
+import { InfiniteScrollDatasource } from './infinite-scroll-datasource';
+
+class TestData {
+    testId: number;
+    testDescription: string;
+
+    constructor(input?: Partial<TestData>) {
+        if (input) {
+            Object.assign(this, input);
+        }
+    }
+}
+
+class TestDataSource extends InfiniteScrollDatasource<TestData> {
+    testDataBatch1: TestData[] = [
+        {
+            testId: 1,
+            testDescription: 'test1'
+        },
+        {
+            testId: 2,
+            testDescription: 'test2'
+        },
+        {
+            testId: 3,
+            testDescription: 'test3'
+        },
+        {
+            testId: 4,
+            testDescription: 'test4'
+        }
+    ];
+    testDataBatch2: TestData[] = [
+        {
+            testId: 5,
+            testDescription: 'test5'
+        },
+        {
+            testId: 6,
+            testDescription: 'test6'
+        }
+    ];
+
+    getNextBatch(pagingOptions: ContentPagingQuery): Observable<TestData[]> {
+        if (pagingOptions.skipCount === 4) {
+            return from([this.testDataBatch2]);
+        } else if (pagingOptions.skipCount === 0) {
+            return from([this.testDataBatch1]);
+        } else {
+            return from([]);
+        }
+    }
+}
+
+@Component({
+    template: ` <cdk-virtual-scroll-viewport appendOnly itemSize="300" style="height: 500px; width: 100%;">
+        <div *cdkVirtualFor="let item of testDatasource" class="test-item" style="display: block; height: 100%; width: 100%;">
+            {{ item.testDescription }}
+        </div>
+    </cdk-virtual-scroll-viewport>`
+})
+class TestComponent implements OnInit {
+    testDatasource = new TestDataSource();
+
+    ngOnInit() {
+        this.testDatasource.batchSize = 4;
+    }
+}
+
+describe('InfiniteScrollDatasource', () => {
+    let fixture: ComponentFixture<TestComponent>;
+    let component: TestComponent;
+
+    const getRenderedItems = (): HTMLDivElement[] => {
```

Comment:
ESlint issue: 
<img width="1393" alt="image" src="https://github.com/Alfresco/alfresco-ng2-components/assets/115619721/306e7375-2b4d-4d1e-b3fd-55bfd7be7139">

Reply:
Fixed

Reply:
@MichalKinas I unresolved this comment because I can see still eslint issue. Probably it is because you have one line arrow function so you can remove {}. You removed () around parameter. 

<img width="1336" alt="image" src="https://github.com/Alfresco/alfresco-ng2-components/assets/115619721/5bbeaced-a39f-4178-b79a-81f482f40bfd">

Reply:
Ah right sorry I misunderstood that, fixed now :D

Reply:
no problem :)